File: backend/src/main/java/com/project/entities/BaseEntity.java
package com.project.entities;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import jakarta.persistence.Column;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@MappedSuperclass
//Lombok annotations
@Getter
@Setter
@ToString
public abstract class BaseEntity {

    
    @CreationTimestamp
	@Column(name="created_on",updatable = false)
	private LocalDateTime createdAt;
	@UpdateTimestamp
	@Column(name="last_updated")
	private LocalDateTime updatedAt;
}

File: backend_dotnet/SafeMind.Core/Entities/BaseEntity.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace SafeMind.Core.Entities
{
    public abstract class BaseEntity
    {
        [Column("created_on")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("last_updated")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
